plugins/tpm-eventlog/fu-plugin-tpm-eventlog.c

```diff
@@ -125,19 +125,22 @@ fu_plugin_device_registered (FuPlugin *plugin, FuDevice *device)
 				     "please see https://github.com/fwupd/fwupd/wiki/TPM-PCR0-differs-from-reconstruction");
 }
 
-gboolean
-fu_plugin_add_security_attrs (FuPlugin *plugin, GPtrArray *attrs, GError **error)
+void
+fu_plugin_add_security_attrs (FuPlugin *plugin, FuSecurityAttrs *attrs)
 {
 	FuPluginData *data = fu_plugin_get_data (plugin);
-	FwupdSecurityAttr *attr = fwupd_security_attr_new ("org.trustedcomputinggroup.TpmEventLog");
+	g_autoptr(FwupdSecurityAttr) attr = NULL;
+
+	attr = fwupd_security_attr_new ("org.trustedcomputinggroup.TpmEventLog");
 	fwupd_security_attr_set_level (attr, FWUPD_SECURITY_ATTR_LEVEL_IMPORTANT);
 	fwupd_security_attr_set_name (attr, "TPM Reconstruction");
-	if (data->reconstructed) {
+	if (!fu_plugin_get_enabled (plugin)) {
```

Comment:
Ah OK, so this lets the plugin vfunc run no matter enabled or not and it can make a decision then what to do with HSI .  I think this is a better approach indeed.